server.go

```diff
@@ -1977,6 +1977,12 @@ func (s *Server) ExecuteQuery(q *influxql.Query, database string, user *User) Re
 
 // executeSelectStatement plans and executes a select statement against a database.
 func (s *Server) executeSelectStatement(stmt *influxql.SelectStatement, database string, user *User) *Result {
+	// Perform any necessary query re-writing.
+	stmt, err := s.rewriteSelectStatement(stmt)
```

Comment:
We should only rewrite if the statement actually has a wildcard. The statement struct should have a method that indicates if it has a wildcard.